BLSQ/iaso
# IA-1648 Generalize xls form generation from polio campaign to iaso globally

Explain what problem this PR is resolving :
Export a xlsform by using a group as OU source. A template is required. Specific data can be extracted from campaigns by
starting the variable name by 'insert_' in the xls file in "calculate" row. The data will be saved in the
calculation column. By adding campaignid to the params you can extract data from a campaign.

## Self proof reading checklist



## How to test
The url to test it with extracting data from a campaign looks like this 
/api/generate_xlsform/xlsform_generator/?groupid=169&form_name=test_form&campaignid=022ca298-b24c-479b-ad81-34a4a6e9782b&ouhierarchy=COUNTRY,REGION,district&outypelist=country,region,district

- groupid  : It's the group of org units that we want to use to generate the xls form. The user can only access the groups linked to his account. 
- form_name : it's the template form name.
- campaignid : the campaign id must be specified in order to extract data from a campaign
- ouhierarchy : provide a list where i[0] is the top parent i[1] is the children i[2] and so on...
- outypelist: list of the type of orgunit we want in the xlsform. 

A xls form template is required.  The xls must contains the basics required field of an xlsform. The choice sheet will be generated. 

The column name for the survey sheet are :
type | name | label | required | appearance | constraint | constraint_message | relevant | hint | choice_filter | jr:count | repeat_count | calculation

Check below for a video on how to test it. 



https://user-images.githubusercontent.com/45785235/204535954-27bed13e-9455-41f0-9c91-74ed2c1b6001.mov
